### DISCUSSION TITLE
Copilot should be free

### DISCUSSION THREAD

**Discussion by contingency1111** [community/community#6859] [Copilot Conversations]
Honestly, I don’t think Github Copilot should be paid. It uses other developers code which is public and open source but you would pay money to use it?

**Comment by TarunavBA** [comment]
I also agree with you @MarieJoker what u say is 💯% true. Well said

**Comment by TarunavBA** [comment]
As it is in the technical preview we are getting it for free if it is released they (@github team) shouldnt charge us and they should also give it to individual developers like us.

**Comment by rvenugopal** [comment]
It is using https://beta.openai.com/pricing
As you can see it is not free for github so I am presuming they will start charging for it eventually unless they plan to use it to keep developers locked in

**Reply by 143mailliw** [reply]
MS has an exclusive license to host Codex and GPT-3 on their servers.

**Reply by MulleDK19** [reply]
> It is using https://beta.openai.com/pricing As you can see it is not free for github so I am presuming they will start charging for it eventually unless they plan to use it to keep developers locked in

Microsoft owns both GitHub and GPT-3, so OpenAI's pricing page means nothing.

**Comment by figloalds** [comment]
Like it or not it's a cloud service as much as OneDrive or GitHub itself is and it has some hardware expense to run
It MAY have some free tier, but it's inevitable that it's going to cost something
I'm just hoping it's a fair/affordable price

**Comment by Studio-Hawaii** [comment]
open-ai does cost money and that's what copilot is using
I do agree though it should be free maybe GitHub could ask for donations instead or something

**Comment by GoAwayNow** [comment]
If it's not free, it could completely destroy freelance developers. It could create a cycle of needing the money for copilot in order to get jobs in order to make more money in order to keep copilot.

**Comment by SimpleCppCoder** [comment]
While I really hope Github Copilot is free, (and it should be) but with the option to add a donation or let there be a paid version that adds (not takes away) something that people would want to buy or to support this tool. Will there be student pricing? I can't remember the name but there was something that gave their service to students for free even after University or Collage.

I completely agree with @MarieJoker . Maybe github copilot should be free to those (I mean the current github accounts linked to copilot, maybe not the waitlist) to those who are testing. I am still on the waitlist, but hopefully I can join soon.
But the thing about using developer's data:

> _Please note that the usage information may include snippets of code that you use, create, or generate while using GitHub Copilot._ When you edit files with the GitHub Copilot plugin enabled, file content snippets and suggestion results will be shared with GitHub and OpenAI and used for diagnostic purposes and to improve suggestions. GitHub Copilot relies on file content, for context, both in the file you are editing and potentially other files in the same Visual Studio Code workspace. **GitHub Copilot does not use your private code as input to suggest code for other users of GitHub Copilot. The code snippets are treated as confidential information and accessed on a need-to-know basis.**

I also agree with @GoAwayNow that freelance or just developers in general would need github copilot to improve their code, or at least speed up the time to write that code. People would rather pay someone who could write good code quickly rather than someone who doesn't use github copilot.
If github copilot does get better and better over the years because of the OpenAI thing that could also replace artists. Creating code or art might just mean paying $20/month, telling the computer to make a new operating system or telling the computer to draw a house and selling that; that would be what developers and artists will turn into.[ Someone who tells a computer what to do.](https://www.commitstrip.com/wp-content/uploads/2016/08/Strip-Les-specs-cest-du-code-650-finalenglish.jpg)

**Comment by Pro100rus32** [comment]
They **run a neural network model on their servers**, and **this requires money**, especially at the current price for video cards.

Perhaps you would be interested in a **local** `copilot` model, but there is already `tabnine`, which **is available to everyone**, **free** and works on your PC.

**Comment by adminvns** [comment]
Man!! I was using GitHub Co-pilot from many days, and just now I got a notification that states "Start your free trial", it should be free @github-staff @github

Please make this feature free maybe you can limit the usage (maybe api request per day/month) then unlimited for paid versions
🙏🙏🙏🙏🙏

**Reply by rauzerfabiano** [reply]
Exactly

**Reply by SimpleCppCoder** [reply]
Yeah, I finally was able to join the testers just a day ago and I was able to use it for five minutes before bed, the next day (today) its gone...

**Reply by HitomiXz** [reply]
sure

**Comment by fornasierov** [comment]
@MarieJoker, I agree. Especially when one thinks of devs in emerging countries. This is a feature that will without a doubt become an industry standard soon, so they should think carefully about the impacts their business model will create in people's lives around the whole world.

**Comment by localfossa** [comment]
They should keep it free.

Edit: Maybe they can keep it free for who applied and joined technical preview.

**Reply by rauzerfabiano** [reply]
I think it's unlikely, but I'm really rooting for it.

**Comment by 201flaviosilva** [comment]
I agree, make no sense to be paid. The GitHub Copilot learned from open source projects, so make no sense we need to paid for this!!

**Comment by supunsathsara** [comment]
@alero-3 It's still free (hope it'll stay that way) for **students** and for maintainers of popular open source projects. 
<br>
![image](https://user-images.githubusercontent.com/67389877/174872226-c7d39333-4a63-4b1d-ad65-a8e761f23f77.png)

**Reply by elevatebart** [reply]
What does it take to be considered an open-source maintainer?

I maintain vue-styleguidist, the engine inside storybookjs for vue and vue-live, but I got asked to pay. 
I feel like I am asked to pay for my own code.

I do not like it one bit.

Did you have to do something specific to get that window?

Thank you so much for being a maintainer.
Bart

**Reply by supunsathsara** [reply]
Sorry, but I got it as a student through GitHub Global Campus.

But according to GitHub, 
 `A maintainer of a popular open source project is defined as someone who has write or admin access to one or more of the most popular open source projects on GitHub` & if u're eligible you should be able to add GitHub Copilot for no charge in the [Copilot subscription page.](https://github.com/github-copilot/free_signup)
 check out [Copilot FAQ](https://github.com/pricing#faq-copilot)
 
 I don't like this subscription based thing too.. They should've add a different pro plan with some extra features while keeping this version free.

**Comment by NeuralAIM** [comment]
Hello from **Russia**!

![image](https://user-images.githubusercontent.com/75082388/174877639-2fde1e19-159d-49cf-aa1d-44c371b06f6d.png)

**Comment by secretgspot** [comment]
How do you expect Bill Gates buy out the rest of the farms if it's free and exists because of our data/sweat

**Reply by rauzerfabiano** [reply]
looool exactly

**Comment by litt1e-p** [comment]
it is time to quit copilot

**Reply by rauzerfabiano** [reply]
Now we've done all the work... millions of lines of code...

**Comment by Hyperclaw79** [comment]
While I do understand the cost required to maintain the servers, it still feels ethically wrong to charge users after training on their code.
I think the best solution for both users and GitHub would be to make Copilot free for all open source projects. Otherwise its against the very essence of open source.
Github, which is a platform that has been enabling developers to be so much more productive, shouldn't make moves which are against its core values.

So yeah, make it free for open source projects. I know it's technically possible to implement it this way.

**Reply by elevatebart** [reply]
I agree with the sentiment and am really hoping that this can be done.

In the meantime, it is mostly free for contributors and maintainers of OSS and for students. I hope it stays that way.

**Reply by Hyperclaw79** [reply]
Nope, not quite unfortunately:
1. Not free for contributors, only the maintainers - the authors and those with admin access to the repos. Imagine a scenario where a repo started off small but got tons of PRs and got popular. Now only a select few users will get free access, but all the others who contributed to the repo will not be considered at all. There are cases where external PRs have been gamechangers for a repo.
2. There's no indication about the threshold for considering you an eligible Maintainer, just a vague message.
3. Not all students get the free access, only the students who have access to the Github Student Pack and this will not be valid anymore after they graduate.

**Comment by Rand-Random** [comment]
Another reason why it should be free and open source 
https://sfconservancy.org/blog/2022/jun/30/give-up-github-launch/

**Comment by antoinekm** [comment]
Speaking of this:

> _I said this last year when Copilot was announced. I think it’s gross that a multi-billion dollar company took all the source code we gave them for free, analyzed it, built a tool, and charges $100 a year for it. Anyone who made a commit to GitHub should get this for free._

Source: https://twitter.com/bphogan/status/1539298492887203840

**Comment by TryTryAgain** [comment]
What qualifies as "popular"? Is that only based on stars for a project, badges, accepted PRs, depending on more than one project or only one is allowed as well? I would like not pay for the full year if I will qualify for free 12 months credits in a few months.

**Reply by MulleDK19** [reply]
It's intentionally vague. And it's not enough to be maintainer on a popular repository, you also have to be the admin.

**Comment by evan-hwang** [comment]
If you don't provide your own local for resources to turn around the copilot, I think it's natural to charge for resources.

**Reply by Hyperclaw79** [reply]
The thing is... The amount they're charging is much higher than what it would cost them to maintain the resources. They're making a profit out of the open source code which is against the philosophy of FOSS.

**Comment by cmp-nct** [comment]
The Codex models are open source as far as I know. So you can train one on github codes and release your model for free.
But the idea that other companies need to do that for free is ridiculous, it costs an enormous amount of computing resources to provide the completions within short delays.

**Comment by mtrojahn** [comment]
It should just come for free for whoever is already a GitHub PRO user or, at most, have another plan for PRO that includes Copilot for the 10 bucks. I mean, the price of Copilot is a bit too much and they could easily add PRO to it. The average developer has no compelling reason to get any of them.
